File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix_list = [
            list(map(int, row.split())) for row in matrix_string.split("\n")
        ]

    def row(self, index):
        return self.matrix_list[index - 1]

    def column(self, index):
        return [i[index - 1] for i in self.matrix_list]

File: javascript/matrix/matrix.js
const getColumnsFromRows = (rows) => {
  return rows[0].map((_, i) => rows.map((row) => row[i]))
}

export class Matrix {
  constructor(matrix) {
    this._rows = matrix.split("\n").map((row) => row.split(" ").map(Number));
    this._columns = getColumnsFromRows(this._rows);
  }

  get rows() {
    return this._rows;
  }

  get columns() {
    return this._columns;
  }
}
